src/main/java/org/folio/circulation/resources/RenewalResource.java

```diff
@@ -73,14 +75,29 @@ private void renew(RoutingContext routingContext) {
       .thenCompose(r -> r.combineAfter(configurationRepository::findTimeZoneConfiguration,
         LoanAndRelatedRecords::withTimeZone))
       .thenComposeAsync(r -> r.after(records -> renewalStrategy.renew(records, bodyAsJson, clients)))
-      .thenComposeAsync(r -> r.after(loanRepository::updateLoan))
+      .thenComposeAsync(r -> r.after(relatedRecords -> updateLoanAndItemInStorage(relatedRecords,
+        itemRepository, loanRepository)))
       .thenApply(r -> r.next(scheduledNoticeService::rescheduleDueDateNotices))
       .thenApply(r -> r.next(loanNoticeSender::sendRenewalPatronNotice))
       .thenApply(r -> r.map(loanRepresentation::extendedLoan))
       .thenApply(LoanResponse::from)
       .thenAccept(result -> result.writeTo(routingContext.response()));
   }
 
+  private CompletableFuture<Result<LoanAndRelatedRecords>> updateLoanAndItemInStorage(
+    LoanAndRelatedRecords relatedRecords, ItemRepository itemRepository,
+    LoanRepository loanRepository) {
+
+    Loan loan = relatedRecords.getLoan();
+    if (loan == null || loan.getItem() == null) {
+      return null;
+    }
+
+    return itemRepository.updateItem(loan.getItem())
```

Comment:
Maybe we should check first whether the item was changed or not: 
```
if (item.hasChanged()) {
  ...
}
```

Can we execute item update and loan update asynchronously?